### PR TITLE
DEV: Resolve API deprecations and modernize component

### PR DESCRIPTION
**Pull request by Arkshine**
This PR does essentially the following:

* Moves the code to an initializer file and remove others unused files
* Resolves the API deprecations by using `decorateCookedElement` and `addComposerToolbarPopupMenuOption`
* Normalizes the settings name (migrations included)
  * `Content_height` -> `scrollable_content_height`,
  * `Scrollable_content_button_icon` -> `scrollable_content_button_icon`,
* Adjusts CSS to support RTL
* Adds basic acceptance tests
* Pins theme component to 3.2.0.beta3 
* Add lint config files

Hopefully I did not miss anything!

https://github.com/discourse/Scrollable-post-content/assets/360640/90a9600b-0e64-4937-9f7e-76409bafdb40

### CODE REVIEW

File: javascripts/discourse/api-initializers/scrollable-post-content.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    { id: "scrollable-post-content" }
+  );
+
+  api.addComposerToolbarPopupMenuOption({
+    action: (toolbarEvent) => {
+      if (!toolbarEvent) {
+        return;
+      }
+      console.log("toolbarEvent", toolbarEvent);
```
Comment (davidtaylorhq) [inline javascripts/discourse/api-initializers/scrollable-post-content.js:36]: Looks like a debugging thing, shall we drop it?

```suggestion
```

**Comment by Arkshine** [comment]
My bad! How could I have missed this? Thanks for the review! 
This should be fixed now. 

image

File: javascripts/discourse/api-initializers/scrollable-post-content.js (hunk 1 above)
Comment (Arkshine) [reply javascripts/discourse/api-initializers/scrollable-post-content.js]: Fixed!

File: locales/en.yml [hunk 2]
```diff
@@ -1,2 +1,9 @@
 en:
+  theme_metadata:
+    description: "Adds scrollable sections in posts with one click!"
+    settings:
+      Content_height:
+        description: "Set the desired height for scollable content in pixles - e.g 300 - Units are not needed."
+      Scrollable_content_button_icon:
```
[the remark this answers is not in the document]
Comment (Arkshine) [reply locales/en.yml]: Fixed!

**Review by davidtaylorhq — APPROVED** [review]
Thanks @Arkshine

**Pull request merged**